environments/pull-environments.sh

```diff
@@ -2,50 +2,23 @@
 set -e
 IFS='|'
 
-# In development, AWS_PROFILE should be set. In CI, it's not.
-[ "$AWS_PROFILE" ] && useProfile="true" || useProfile="false";
-
-FRONTENDCONFIG="{\
-\"SourceDir\":\"src\",\
-\"DistributionDir\":\"dist\",\
-\"BuildCommand\":\"npm run-script build\",\
-\"StartCommand\":\"npm run-script start\"\
-}"
-FRONTEND="{\
-\"frontend\":\"javascript\",\
-\"framework\":\"none\",\
-\"config\":$FRONTENDCONFIG\
-}"
-AMPLIFY="{\
-\"defaultEditor\":\"code\",\
-}"
-AWSCLOUDFORMATIONCONFIG="{\
-\"configLevel\":\"project\",\
-\"useProfile\":$useProfile,\
-\"profileName\":\"$AWS_PROFILE\",\
-\"accessKeyId\":\"$AWS_ACCESS_KEY_ID\",\
-\"secretAccessKey\":\"$AWS_SECRET_ACCESS_KEY\",\
-\"region\":\"us-east-1\"\
-}"
-PROVIDERS="{\
-\"awscloudformation\":$AWSCLOUDFORMATIONCONFIG\
-}"
-
+dirs=""
 # Pull the backend for each environment
 for dir in ./*/ ; do
   if ! [ -f "$dir/package.json" ]; then
     echo "If $dir is an environment, ensure the a package.json file exists with a \"pull\" command that pulls the environment (see the README)."
     continue
   fi
 
-  cd $dir
+  dirs="$dirs\n$dir"
+done
+
+# max number of parallel tasks at a time
+numParallelTasks=8; # Future improvement: could set this to # of logical cores in localdevice
 
-  # 'echo y' is used to answer "Yes" to the prompt "Do you plan on modifying this backend?"
-  # See https://github.com/aws-amplify/amplify-cli/issues/5275
-  echo y | yarn pull \
-    --amplify $AMPLIFY \
-    --frontend $FRONTEND \
-    --providers $PROVIDERS
+if [ "$NODE_ENV" = "test" ]; then
+  numParallelTasks=1; # GitHub actions has trouble handling parallel executions
```

Comment:
😢  See PR body -- this was failing, with inconsistent errors, whenever we run parallel pulls. I might come back to this on my Linux machine. But considering actions runner only has two cores, this might not be worth the extra effort.

Reply:
I'm down to merge this for the local DX benefit, even though it's not realized in CI.